components/monitoring/grafana/base/grafana-app.yaml

```diff
@@ -15,52 +29,175 @@ metadata:
     kubernetes.io/service-account.name: metrics-reader
 type: kubernetes.io/service-account-token
 ---
-# Create cluster role binding
-kind: ClusterRoleBinding
+# Create service account to connect to Grafana
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: grafana-serviceaccount
+  namespace: appstudio-grafana
+  annotations:
+    argocd.argoproj.io/compare-options: IgnoreExtraneous
+---
+# Create secret for the Grafana service account
+apiVersion: v1
+kind: Secret
+metadata:
+  name: grafana-proxy
+  namespace: appstudio-grafana
+  annotations:
+    kubernetes.io/service-account.name: grafana-serviceaccount
+type: kubernetes.io/service-account-token
+---
+# Create Cluster Role for the Grafana service account
 apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
 metadata:
-  name: cluster-monitoring-view-grafana
+  name: grafana-proxy
+rules:
+  - verbs:
+      - create
+    apiGroups:
+      - authentication.k8s.io
+    resources:
+      - tokenreviews
+  - verbs:
+      - create
+    apiGroups:
+      - authorization.k8s.io
+    resources:
+      - subjectaccessreviews
+---
+# Create a CRB for the Grafana service account
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  namespace: appstudio-grafana
+  name: grafana-proxy
 subjects:
   - kind: ServiceAccount
-    name: metrics-reader
+    name: grafana-serviceaccount
     namespace: appstudio-grafana
 roleRef:
-  apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
-  name: cluster-monitoring-view
+  name: grafana-proxy
+  apiGroup: rbac.authorization.k8s.io
 ---
-# Create a Grafana instance
-# Do not add security to this yaml
+# Add Config Map to inject certificates
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: appstudio-grafana
+  labels:
+    config.openshift.io/inject-trusted-cabundle: "true"
+  name: ocp-injected-certs
+---
+# Create the Grafana instance
 apiVersion: integreatly.org/v1alpha1
 kind: Grafana
 metadata:
-  annotations:
-    argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
-  name: appstudio-grafana
   namespace: appstudio-grafana
+  name: grafana-oauth
 spec:
   config:
-    auth:
-      disable_signout_menu: true
-    auth.anonymous:
-      enabled: true
+    users:
+      viewers_can_edit: true
     log:
-      level: warn
-      mode: console
+      mode: "console"
+      level: "warn"
+    auth.anonymous:
+      enabled: True
+    auth:
+      disable_login_form: False
+      disable_signout_menu: True
+    auth.basic:
+      enabled: True
```

Comment:
Is this enabled so the auth proxy can use it?

Reply:
Yes, we need it for internal use of Grafana.
It was suggested to us by the Grafana Operator contributors